app/[locale]/faculty-and-staff/page.tsx

```diff
@@ -72,134 +52,65 @@ export default async function FacultyAndStaff({
   });
-  const currentDepartment = departments.find(
-    ({ urlName }) => urlName === department
-  );
 
-  return select ? (
-    <Select
-      defaultValue={currentDepartment && `?department=${department}`}
-      navigate
-    >
-      <SelectTrigger className="px-4 py-5 sm:w-1/2 lg:w-1/3 xl:hidden">
-        <SelectValue placeholder="Choose a department" />
-      </SelectTrigger>
-      <SelectContent>
-        {departments.map(({ name, urlName }, index) => (
-          <SelectItem key={index} value={`?department=${urlName}`}>
-            {name}
-          </SelectItem>
-        ))}
-      </SelectContent>
-    </Select>
-  ) : (
-    <ol className="w-full space-y-4">
-      {departments.map(({ name, urlName }, index) => (
-        <li key={index}>
-          <Button
-            active={urlName === currentDepartment?.urlName}
-            asChild
-            className="font-semibold text-shade-dark"
-            variant="link"
-          >
-            <Link href={{ query: { department: urlName } }}>{name}</Link>
-          </Button>
-        </li>
-      ))}
-    </ol>
-  );
+  return <Tabs select={select} departments={departments} />;
 };
 
 const FacultyList = async ({
-  department,
   deptartmentHeadText,
   locale,
-  query,
 }: {
-  department?: string;
   deptartmentHeadText: string;
   locale: string;
-  query?: string;
 }) => {
-  const departments = await db.query.departments.findMany({
-    columns: { id: true, name: true, urlName: true },
-  });
-  const departmentHeads = await db.query.departmentHeads.findMany({
-    where: (departmentHead, { eq }) => eq(departmentHead.isActive, true),
-  });
-  const currentDepartment = departments.find(
-    ({ urlName }) => urlName === department
-  );
+  const departments = await db.query.departments
+    .findMany({
+      columns: { id: true, urlName: true },
+    })
+    .then((departments) =>
+      departments.reduce((map, department) => {
+        map.set(department.urlName, department.id);
+        return map;
+      }, new Map<string, number>())
+    );
+
+  const departmentHeads = await db.query.departmentHeads
+    .findMany({
+      where: (departmentHead, { eq }) => eq(departmentHead.isActive, true),
+      columns: { facultyId: true },
+    })
+    .then((departmentHeads) =>
+      departmentHeads.reduce((set, departmentHead) => {
+        set.add(departmentHead.facultyId);
+        return set;
+      }, new Set<number>())
+    );
+
+  //const currentDepartmentId = departments.get(department ?? '');
 
   const faculty = await db.query.faculty.findMany({
     columns: {
       designation: true,
       employeeId: true,
       id: true,
+      departmentId: true,
     },
-    where: currentDepartment
-      ? (faculty, { eq }) => eq(faculty.departmentId, currentDepartment.id)
-      : undefined,
+    // where: currentDepartment
+    //   ? (faculty, { eq }) => eq(faculty.departmentId, currentDepartment.id)
+    //   : undefined,
```

Comment:
I think I was unclear in the issue. I think it's better to keep departments server side and just the query part client side.

Reply:
why, select's already client, its just an extra div. If the data's already there. I don't see any reason to visit the server for no reason.

Reply:
Huh? The difference here is that your code will fetch faculty from all departments than just one department; it isn't just one extra div, we're visiting the server for getting a subset of the data in faculty for faster response times.

Reply:
but wont this get cached anyway?